### PR TITLE
Add support for signature option to url()

### PR DESCRIPTION
**Pull request by maoznir**

### CODE REVIEW

File: src/url.js [hunk 1]
```diff
@@ -156,68 +289,38 @@ function finalizeResourceType(resourceType = "image", type = "upload", urlSuffix
...
  *  For details on all fetch types, see
  * <a href="https://cloudinary.com/documentation/image_transformations#fetching_images_from_remote_locations"
- *  target="_blank">Fetch types</a>. 
- * @param {Object} [options.resource_type='image'] - The type of asset. <p>Possible values:<br/> 
+ *  target="_blank">Fetch types</a>.
+ * @param {Object} [options.resource_type='image'] - The type of asset. <p>Possible values:<br/>
  *  - `image`<br/>
  *  - `video`<br/>
- *  - `raw` 
+ *  - `raw`
```
Comment (asisayag2) [inline src/url.js]: @maoznir, please update the doc string

File: src/url.js [hunk 2]
```diff
@@ -156,68 +289,38 @@ function finalizeResourceType(resourceType = "image", type = "upload", urlSuffix
...
   if (options instanceof Transformation) {
     options = options.toOptions();
   }
+
   options = defaults({}, options, config, DEFAULT_IMAGE_PARAMS);
+
   if (options.type === 'fetch') {
     options.fetch_format = options.fetch_format || options.format;
     publicId = absolutize(publicId);
   }
```
Comment (asisayag2) [inline src/url.js]: @maoznir , would extract that to a "prepareOptions" or "handleOptions" method

File: src/url.js [hunk 3]
```diff
@@ -146,6 +159,126 @@ function finalizeResourceType(resourceType = "image", type = "upload", urlSuffix
...
+/**
+ * Get version part of the url
+ * @param publicId
+ * @param options
+ * @returns {string}
+ */
+function handleVersion(publicId, options) {
+  const isForceVersion = (options.force_version || typeof options.force_version === 'undefined');
+  const isPublicIdIncludesVersion = publicId.search('/') < 0 || publicId.match(/^v[0-9]+/) || publicId.match(/^https?:\//);
+  const isVersionSet = options.version && options.version.toString();
```
Comment (asisayag2) [inline src/url.js]: @maoznir , please add some explanation about the logics here.. what does each section check?

File: src/url.js (hunk 2 above)
Comment (maoznir) [reply src/url.js]: done

File: src/url.js (hunk 1 above)
Comment (maoznir) [reply src/url.js]: in what way?

File: src/url.js (hunk 3 above)
Comment (maoznir) [reply src/url.js]: done

File: src/url.js (hunk 1 above)
Comment (maoznir) [reply src/url.js]: done

Comment (asisayag2) [reply src/url.js]: @maoznir , is type and resource_type sent separately to this method?

File: src/url.js (hunk 2 above)
Comment (asisayag2) [reply src/url.js]: @maoznir Would break the handling of options and publicId (even if we ask "if type==fetch" twice)

File: src/url.js (hunk 3 above)
Comment (asisayag2) [reply src/url.js]: @maoznir , thanks! much clearer

File: src/url.js (hunk 2 above)
Comment (maoznir) [reply src/url.js]: done

File: src/url.js [hunk 4]
```diff
@@ -282,7 +287,7 @@ function urlString(publicId, options) {
 
 /**
  * Merge options and config with defaults
- * update options and publicId according to 'fetch' param
+ * update options fetch_format according to 'type' param
  * @param publicId
```
Comment (asisayag2) [inline src/url.js]: publicId is not used

Comment (maoznir) [reply src/url.js]: you're right, fixed

**Pull request merged**